evm/send-tokens.js

```diff
@@ -2,25 +2,50 @@
 
 require('dotenv').config();
 
-const { ethers } = require('hardhat');
-const { Wallet, getDefaultProvider } = ethers;
-const { Command, Option } = require('commander');
 const chalk = require('chalk');
-const { printInfo, printWalletInfo } = require('./utils');
+const { Command, Option } = require('commander');
+const { ethers } = require('hardhat');
+const {
+    getDefaultProvider,
+    utils: { parseEther },
+} = ethers;
 const readlineSync = require('readline-sync');
 
-async function sendTokens(chain, options) {
+const { printInfo, printWalletInfo, isValidNumber, saveConfig } = require('./utils');
+const {
+    getAllSignersData,
+    updateSignersData,
+    getTransactions,
+    getWallet,
+    ledgerSign,
+    getLocalNonce,
+    updateLocalNonce,
+} = require('./offline-sign-utils.js');
+
+async function sendTokens(config, chain, options) {
+    const { privateKey, offline, ledgerPath, filePath, nonceOffset, env } = options;
+    let { amount, recipients } = options;
     const provider = getDefaultProvider(chain.rpc);
-    const wallet = new Wallet(options.privateKey, provider);
+    recipients = options.recipients.split(',').map((str) => str.trim());
+    amount = parseEther(amount);
 
-    const balance = await printWalletInfo(wallet);
-    const amount = ethers.utils.parseEther(options.amount);
+    const { wallet, providerNonce } = await getWallet(privateKey, provider, ledgerPath);
+    const signerAddress = await wallet.getAddress();
+    let nonce = getLocalNonce(chain, signerAddress);
 
-    if (balance.lte(amount)) {
-        throw new Error(`Wallet has insufficient funds.`);
+    if (providerNonce !== undefined && providerNonce !== null && providerNonce > nonce) {
+        config.chains[chain.name.toLowerCase()] = updateLocalNonce(chain, providerNonce, signerAddress);
```

Comment:
let's avoid updating nonces as part of the tx generation script. offline wallets won't be used online (since the benefit of keeping it offline is lost when connected online). As part of the process, update-nonces should be run before tx creation + sign